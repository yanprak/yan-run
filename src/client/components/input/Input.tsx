import React from 'react';
import { Props } from './types';
import './input.scss';

function Input({ title, errorMessage, ...props }: Props) {
  return (
    <label className="input">
      {title && <span className="input__title">{title}</span>}
      <input
        className="input__control"
        autoComplete="off"
        {...props}
      />
      {errorMessage && <span className="input__error">{errorMessage}</span>}
    </label>
  );
}
export default Input;
