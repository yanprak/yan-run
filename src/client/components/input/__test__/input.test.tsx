import React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from '../Input';

test('Input test', () => {
  const component = TestRenderer.create(
    <Input title="Name" errorMessage="Error text" name="name" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
