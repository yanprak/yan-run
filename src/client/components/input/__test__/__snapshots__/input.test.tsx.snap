// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test 1`] = `
<label
  className="input"
>
  <span
    className="input__title"
  >
    Name
  </span>
  <input
    autoComplete="off"
    className="input__control"
    name="name"
  />
  <span
    className="input__error"
  >
    Error text
  </span>
</label>
`;
